include/cuco/detail/hash_functions/murmurhash3.cuh

```diff
@@ -139,9 +141,24 @@ struct MurmurHash3_32 {
    */
   constexpr result_type __host__ __device__ operator()(Key const& key) const noexcept
   {
-    constexpr int len         = sizeof(argument_type);
+    return (*this)(key, cuco::experimental::extent<std::size_t, sizeof(Key)>{});
+  }
+
+  /**
+   * @brief Returns a hash value for its argument, as a value of type `result_type`.
+   *
+   * @tparam Extent The extent type
+   *
+   * @param key The input argument to hash
+   * @param size The extent of the key in bytes
+   * @return A resulting hash value for `key`
+   */
+  template <typename Extent>
+  constexpr result_type __host__ __device__ operator()(Key const& key, Extent size) const noexcept
```

Comment:
As discussed offline. We would provide a member like
```cuda
 constexpr result_type __host__ __device__ compute_hash(Key const& key, Extent size) const noexcept;
```
instead of overloading `()` operator.

Per[ `std::hash`](https://en.cppreference.com/w/cpp/utility/hash/operator())
```
std::hash<Key>::operator()
 C++ Utilities library std::hash 
Specializations of std::hash should define an operator() that:

Takes a single argument key of type Key.
```

Reply:
Technically we still meet the standard requirements/interface. This is just another overload of the operator. The standard does not prohibit additional functionality.

Reply:
A bit more modern approach would be
```cpp
template<class T, std::size_t N>
constexpr result_type __host__ __device__ compute_hash(cuda::std::span<T, N> data) const noexcept;